tests/unit/lms/services/oauth_http_test.py

```diff
@@ -135,6 +136,96 @@ def test_access_token_request_if_theres_no_refresh_token_or_expires_in(
 
         oauth2_token_service.save.assert_called_once_with("access_token", None, None)
 
+    def test_refresh_token_request(
+        self,
+        svc,
+        http_service,
+        OAuthTokenResponseSchema,
+        oauth_token_response_schema,
+        oauth2_token_service,
+    ):
+        oauth_token_response_schema.parse.return_value = {
+            "access_token": "access_token",
+            "refresh_token": "refresh_token",
+            "expires_in": 1234,
+        }
+
+        svc.refresh_token_request(
+            sentinel.token_url, sentinel.redirect_uri, sentinel.auth
+        )
+
+        http_service.post.assert_called_once_with(
+            sentinel.token_url,
+            data={
+                "grant_type": "refresh_token",
+                "redirect_uri": sentinel.redirect_uri,
+                "refresh_token": oauth2_token_service.get.return_value.refresh_token,
+            },
+            auth=sentinel.auth,
+        )
+        OAuthTokenResponseSchema.assert_called_once_with(http_service.post.return_value)
+        oauth2_token_service.save.assert_called_once_with(
+            "access_token", "refresh_token", 1234
+        )
+
+    def test_refresh_token_raises_if_we_dont_have_a_refresh_token(
+        self, svc, oauth2_token_service
+    ):
+        oauth2_token_service.get.side_effect = OAuth2TokenError
+
+        with pytest.raises(OAuth2TokenError):
+            svc.refresh_token_request(
+                sentinel.token_url, sentinel.redirect_uri, sentinel.auth
+            )
+
+    def test_refresh_token_request_raises_HTTPError_if_the_HTTP_request_fails(
+        self, svc, http_service
+    ):
+        http_service.post.side_effect = HTTPError
+
+        with pytest.raises(HTTPError):
+            svc.refresh_token_request(
+                sentinel.token_url, sentinel.redirect_uri, sentinel.auth
+            )
```

Comment:
Totally hypocritical of me, but I think this gets a pass without a caller as it crams onto nearly one line

Reply:
:man_shrugging: Sure. We do pass exactly the same args to `refresh_token_request()` every time we call it, so I think we could totally add a caller fixture. But I'll hold off